### PR TITLE
chore: add TypeScript styles and norms

### PR DESCRIPTION
**Pull request by cguedes**
Add recommended rules from https://github.com/refstudio/refstudio/issues/18

- 120 char width (prettier)
- Add eslint deps and rules (.eslintrc)
- Fix files using: simple-import-sort
- Manual fixes for any

I've also added the .`/vscode/settings.json` and .`/vscode/extensions.json` with recommended settings. It makes easier to comply with rules after a git clone.

### CODE REVIEW

**Comment by hammer** [comment]
Should we use [pre-commit](https://pre-commit.com/) to automate the enforcement of these guidelines?

**Comment by cguedes** [comment]
@hammer this project was already using (from Tauri) [husky](https://typicode.github.io/husky).
I've fixed the commit-hook and now (after `yarn install`) should be working.

I've configured the project to use, _file based_ commit hook (.husky/pre-commit), to run the already configured `lint-staged`.

**Comment by danvk** [comment]
My two cents: I've always preferred pre-push hooks (rather than pre-commit) so that making local commits is instant but your code is always nicely formatted and linted by the time anyone else would have to look at it.

**Comment by cguedes** [comment]
Added pre-push that runs `lint:check`

File: package.json [hunk 1]
```diff
@@ -37,22 +40,22 @@
...
+    "@typescript-eslint/eslint-plugin": "^5.59.6",
+    "@typescript-eslint/parser": "^5.59.6",
     "@vitejs/plugin-react": "^3.0.0",
-    "husky": "^8.0.3",
+    "eslint": "^8.41.0",
+    "eslint-plugin-react": "^7.32.2",
+    "eslint-plugin-react-hooks": "^4.6.0",
+    "eslint-plugin-simple-import-sort": "^10.0.0",
+    "husky": "^8.0.0",
     "lint-staged": "^13.2.0",
```
Comment (sehyod) [inline package.json:51]: Do we still need lint-staged if we only setup pre-push hooks?

**Review by sehyod — APPROVED** [review]
Looks good to me, thank you for this! I made a comment about `lint-staged`, that I think we can remove, but nothing blocking

**Comment by cguedes** [comment]
This was fixed by #24

**Pull request merged**